File: mixin-padang-webface/src/main/js/padang/model/XVariable.ts
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import EMap from "webface/model/EMap";
import BasicEMap from "webface/model/BasicEMap";
import EAttribute from "webface/model/EAttribute";
import EReference from "webface/model/EReference";
import BasicEObject from "webface/model/BasicEObject";

import * as model from "padang/model/model";
import XPreparation from "padang/model/XPreparation";

export default class XVariable extends BasicEObject {

	public static XCLASSNAME: string = model.getEClassName("XVariable");

	public static FEATURE_NAME = new EAttribute("name", EAttribute.STRING);
	public static FEATURE_FORMULA = new EAttribute("formula", EAttribute.STRING);
	public static FEATURE_PROPERTIES = new EAttribute("properties", EAttribute.STRING);
	public static FEATURE_PREPARATION = new EReference("preparation", XPreparation);

	private name: string = null;
	private formula: string = null;
	private properties: EMap<string> = new BasicEMap<string>(this, XVariable.FEATURE_PROPERTIES);
	private preparation: XPreparation = null;

	constructor() {
		super(model.createEClass(XVariable.XCLASSNAME), [
			XVariable.FEATURE_NAME,
			XVariable.FEATURE_FORMULA,
			XVariable.FEATURE_PROPERTIES,
			XVariable.FEATURE_PREPARATION,
		]);
	}

	public getName(): string {
		return this.name;
	}

	public setName(newName: string) {
		let oldName = this.name;
		this.name = newName;
		this.eSetNotify(XVariable.FEATURE_NAME, oldName, newName);
	}

	public getFormula(): string {
		return this.formula;
	}

	public setFormula(newFormula: string) {
		let oldFormula = this.formula;
		this.formula = newFormula;
		this.eSetNotify(XVariable.FEATURE_FORMULA, oldFormula, newFormula);
	}

	public getProperties(): EMap<string> {
		return this.properties;
	}

	public getPreparation(): XPreparation {
		return this.preparation;
	}

	public setPreparation(newPreparation: XPreparation): void {
		let oldPreparation = this.preparation;
		this.preparation = newPreparation;
		this.eSetNotify(XVariable.FEATURE_PREPARATION, oldPreparation, newPreparation);
	}

}

File: mixin-padang/src/main/java/com/andia/mixin/padang/model/XVariable.java
/*
 * Copyright (c) 2020-2023 Datamixin.
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>. */
package com.andia.mixin.padang.model;

import com.andia.mixin.model.BasicEMap;
import com.andia.mixin.model.BasicEObject;
import com.andia.mixin.model.EAttribute;
import com.andia.mixin.model.EFeature;
import com.andia.mixin.model.EMap;
import com.andia.mixin.model.EReference;

public class XVariable extends BasicEObject {

	public static String XCLASSNAME = Padang.getEClassName("XVariable");

	public static EAttribute FEATURE_NAME = new EAttribute("name", EAttribute.STRING);
	public static EAttribute FEATURE_FORMULA = new EAttribute("formula", EAttribute.STRING);
	public static EAttribute FEATURE_PROPERTIES = new EAttribute("properties", EAttribute.STRING);
	public static EReference FEATURE_PREPARATION = new EReference("preparation", XPreparation.class);

	private String name = null;
	private String formula = null;
	private EMap<String> properties = new BasicEMap<String>(this, XFacet.FEATURE_PROPERTIES);
	private XPreparation preparation = null;

	public XVariable() {
		super(Padang.createEClass(XCLASSNAME), new EFeature[] {
				XVariable.FEATURE_NAME,
				XVariable.FEATURE_FORMULA,
				XVariable.FEATURE_PROPERTIES,
				XVariable.FEATURE_PREPARATION,
		});
	}

	public String getName() {
		return this.name;
	}

	public void setName(String newName) {
		String oldName = this.name;
		this.name = newName;
		this.eSetNotify(FEATURE_NAME, oldName, newName);
	}

	public String getFormula() {
		return this.formula;
	}

	public void setFormula(String newFormula) {
		String oldFormula = this.formula;
		this.formula = newFormula;
		this.eSetNotify(FEATURE_FORMULA, oldFormula, newFormula);
	}

	public EMap<String> getProperties() {
		return properties;
	}

	public XPreparation getPreparation() {
		return preparation;
	}

	public void setPreparation(XPreparation newPreparation) {
		XPreparation oldPreparation = this.preparation;
		this.preparation = newPreparation;
		this.eSetNotify(XVariable.FEATURE_PREPARATION, oldPreparation, newPreparation);
	}

}